### PR TITLE
SpinButton: SpinButton: Preventing and Stopping Propagation when enter is pressed

### PR DESCRIPTION
**Pull request by chang47**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [X] Include a change request file using `$ npm run change`

#### Description of changes

This is a fix for a very niche situation.

If we press enter in the spin button and we move focus to another element which also have event event listeners that listen for for keypress or keyup, the event will also get fired.

We want to prevent this from happening by making sure we prevent default and stop propagation for the event if the user presses enter

#### Focus areas to test

In the spin button page, I validated that all standard behavior still persists:
* Changing the value with arrow keys
* Typing a different text value and pressing enter/tab
* In my specific scenario when pressing enter moves focus outside of the spin button, the other key events don't get fired anymore.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx [hunk 1]
```diff
@@ -409,7 +409,7 @@ export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState
   private _handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>): void => {
     // eat the up and down arrow keys to keep focus in the spinButton
     // (especially when a spinButton is inside of a FocusZone)
-    if (event.which === KeyCodes.up || event.which === KeyCodes.down) {
+    if (event.which === KeyCodes.up || event.which === KeyCodes.down || event.which === KeyCodes.enter) {
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx]: and ENTER still submits the change in the spinButton?

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx]: @chang47 can you confirm this? I don't want to merge until we're sure this functionality is still the same.

Comment (chang47) [reply packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx]: @micahgodbolt Yes it still submits, however I have one concern. I looked at the spin button behavior on the fabric website: https://developer.microsoft.com/en-us/fabric#/components/spinbutton

and I noticed that when I press enter, the SpinButton text gets selected. I thought I caused a regression, but when I removed the code I added, I noticed that the text still doesn't get selected after pressing enter.

I see that the only way for the text to get selected would be in the onFocus() function, so I believe there seems to be some sort of discrepancy.

What is the expected behavior for this?

Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx]: I'm seeing the text get selected when I press ENTER on the fabric website, which is expected.

Did you make sure you didn't have cached js when you tested with your change removed?

The expected behavior is that the text is selected when ENTER is pressed

Comment (chang47) [reply packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx]: I tried reproing this behavior directly from the main fabric repo and the text is still not selected after pressing enter, I believe someone must have introduced a regression from a different PR.

I believe the reason is that we're not calling the onFocus code after we press enter. I don't think me stopping a keyboard event from propagating would stop a focus event. So I think what I will do is merge my change in and create an issue. 

Is that acceptable to you?

**Pull request merged**